components/popover/utils.js

```diff
@@ -0,0 +1,86 @@
+
+/**
+ * Module constants
+ */
+const HEIGHT_OFFSET = 10; // used by the arrow and a bit of empty space
+const isMobileViewport = () => window.innerWidth < 782;
+
+/**
+ * Utilitity used to compute the popover position and the content max width/height for a popover
+ * given its anchor rect and its content size.
+ *
+ * @param {Object} anchorRect       Anchor Rect.
+ * @param {Object} contentSize      Content Size.
+ * @param {string} position         Position.
+ * @param {boolean} expandOnMobile  Whether to expand the popover on mobile or not.
+ *
+ * @return {Object} Popover position and constraints.
+ */
+export function computePopoverPosition( anchorRect, contentSize, position = 'top', expandOnMobile = false ) {
+	const { width, height } = contentSize;
+	const popoverLeft = Math.round( anchorRect.left + ( anchorRect.width / 2 ) );
+	const [ yAxis, xAxis = 'center' ] = position.split( ' ' );
+
+	// y axis aligment choices
+	const topAlignment = {
+		popoverTop: anchorRect.top,
+		contentHeight: anchorRect.top - HEIGHT_OFFSET - height > 0 ? height : anchorRect.top - HEIGHT_OFFSET,
+	};
+	const bottomAlignment = {
+		popoverTop: anchorRect.bottom,
+		contentHeight: anchorRect.bottom + HEIGHT_OFFSET + height > window.innerHeight ? window.innerHeight - HEIGHT_OFFSET - anchorRect.bottom : height,
```

Comment:
Maybe window.innerHeight, and window.innerWidth can be parameters with default values to the window properties. I think it may simplify the addition of test cases in the future.